reactive/flannel.py

```diff
@@ -190,24 +190,40 @@ def configure_network(etcd):
     if port:
         flannel_config['Backend']['Port'] = port
 
-    data = json.dumps(flannel_config)
-    cmd = "etcdctl "
-    cmd += "--endpoint '{0}' ".format(etcd.get_connection_string())
-    cmd += "--cert-file {0} ".format(ETCD_CERT_PATH)
-    cmd += "--key-file {0} ".format(ETCD_KEY_PATH)
-    cmd += "--ca-file {0} ".format(ETCD_CA_PATH)
-    cmd += "set /coreos.com/network/config '{0}'".format(data)
+    output = run_etcdctl(etcd, "get", "/coreos.com/network/config")
+    current_config = json.loads(output)
+    # Check if subnet-len changed in comparison with what's stored in etcd,
+    # but more important than that it's to check for a not set SubnetLen when
+    # the subnet-len configured is 24 which will be the case for environments
+    # upgrading the charm and in those situations we don't want to add
+    # disruption restating flannel without need.
+    if current_config.get("SubnetLen") != config("subnet-len") \
```

Comment:
readability suggestion:

```suggestion
    current_subnet = current_config.get("SubnetLen")
    target_subnet = config("subnet-len")
    if (current_subnet != target_subnet) and (current_subnet is not None) and (target_subnet == 24):
```

@freyes why would you only apply this when `subnet-len` is 24?